packages/nimble-components/src/table/tests/table-keyboard-navigation.spec.ts

```diff
@@ -0,0 +1,1059 @@
+            });
+        });
+
+        describe('for a table with a grouped column, when a group row is focused,', () => {
+            beforeEach(async () => {
+                await setupTableWithGrouping();
+                element.focus();
+                await waitForUpdatesAsync();
+                await sendKeyPressToTable(keyArrowDown);
+                if (!(currentFocusedElement() instanceof TableGroupRow)) {
+                    throw new Error('Expected group row to be focused');
+                }
+            });
+
+            const collapseGroupKeys = [keySpace, keyEnter, keyArrowLeft];
+            collapseGroupKeys.forEach(key => {
+                it(`for an expanded group row, pressing ${key} will collapse the group`, async () => {
+                    await sendKeyPressToTable(key);
+
+                    const focusedElement = currentFocusedElement();
+                    expect(focusedElement).toBeInstanceOf(TableGroupRow);
+                    expect((focusedElement as TableGroupRow).expanded).toBe(
+                        false
+                    );
+                });
+            });
+            const expandGroupKeys = [keySpace, keyEnter, keyArrowRight];
+            expandGroupKeys.forEach(key => {
+                it(`for a collapsed group row, pressing ${key} will expand the group`, async () => {
+                    pageObject.toggleGroupRowExpandedState(0);
+                    await waitForUpdatesAsync();
+                    let focusedElement = currentFocusedElement();
+                    if (
+                        !(focusedElement instanceof TableGroupRow)
+                        || focusedElement.expanded
+                    ) {
+                        throw new Error(
+                            'Expected focused group row to be collapsed'
+                        );
+                    }
+
+                    await sendKeyPressToTable(key);
+
+                    focusedElement = currentFocusedElement();
+                    expect(focusedElement).toBeInstanceOf(TableGroupRow);
+                    expect((focusedElement as TableGroupRow).expanded).toBe(
+                        true
+                    );
+                });
+            });
+
+            it('with no tabbable elements in the group row, pressing Tab will move focus past the table', async () => {
+                const keyEvent = await sendKeyPressToTable(keyTab);
+
+                await verifyLastTabKeyEventBehavior(keyEvent);
+            });
+
+            it('with no tabbable elements in the group row, pressing Shift-Tab will move focus past the table', async () => {
+                const keyEvent = await sendKeyPressToTable(keyTab, {
+                    shiftKey: true
+                });
+
+                await verifyLastTabKeyEventBehavior(keyEvent);
+            });
+        });
+    });
+
+    describe('with anchor columns,', () => {
```

Comment:
What would be your expectation for keyboard navigation test coverage when adding a new column type? Could we do anything now to make it easier to know what tests should be added in the future? Not demanding major refactors, just some hints to prevent you from needing to think through all of this again in a few weeks when we add the menu button column and button column.

For example right now it looks like we have tests for text column and anchor column. Is text providing coverage for all non-interactive columns? If so then maybe we rename its `describe` to "for non-interactive column" and the fact that we use text is an implementation detail. And we could do the same for anchor if it covers cases for all interactive columns. Or if it doesn't then maybe there's a way to tweak this to make it simpler to add tests for the next interactive column. Some brainstorm-y ideas which range from trivial to not trivial:
- add an extra layer of `describe` that doesn't add any new functionality but gives a place for new tests. `describe('with interactive column of type', ... () => ... describe('anchor', ...) )`
- rename the anchor test `it` descriptions to be agnostic to anchors so they can be copied more easily
- parameterize everything about the anchor tests so new columns can just add new values to the parameters array
- a comment saying "new interactive columns should replicate these tests"

Reply:
- Non-interactive columns: I expect that the tests that we have now are sufficient to cover that code path (regardless of if there's new non-interactive column types in the future).
  - Updated wording to be `with non-interactive (text) columns`
- Interactive column types: Good point that it wasn't especially clear how the tests would apply to other column types. I think all the tests I had for the anchor would also apply for other interactive column types. Something like a menu button column would probably have additional tests (because opening a menu is extra functionality that needs to be tested in the context of keyboard navigation). But the current tests there would still be the baseline.
  - Added `describe('with interactive columns of type'`, and an inner `parameterizeSpec` block (that currently only has `'anchor'`, but it should be easy to add additional column types there, we just need to know how to setup the table / what table data / how to query the cell contents))